lib/msal-common/src/client/BaseClient.ts

```diff
@@ -79,85 +63,72 @@ export abstract class BaseClient {
 
         // Set the network interface
         this.networkClient = this.config.networkInterface;
-    }
 
-    // #region Abstract Functions
-
-    /**
-     * Creates a url for logging in a user. This will by default append the client id to the list of scopes,
-     * allowing you to retrieve an id token in the subsequent code exchange. Also performs validation of the request parameters.
-     * Including any SSO parameters (account, sid, login_hint) will short circuit the authentication and allow you to retrieve a code without interaction.
-     * @param request
-     */
-    abstract async createLoginUrl(request: AuthenticationParameters): Promise<string>;
+        // Default authority instance.
+        this.defaultAuthorityInstance = AuthorityFactory.createInstance(
+            this.config.authOptions.authority || Constants.DEFAULT_AUTHORITY,
+            this.networkClient
+        );
+    }
 
     /**
-     * Creates a url for logging in a user. Also performs validation of the request parameters.
-     * Including any SSO parameters (account, sid, login_hint) will short circuit the authentication and allow you to retrieve a code without interaction.
-     * @param request
+     * Create authority instance if not set already, resolve well-known-endpoint
+     * @param authorityString
      */
-    abstract async createAcquireTokenUrl(request: AuthenticationParameters): Promise<string>;
+    protected async createAuthority(authorityString: string): Promise<Authority> {
 
-    /**
-     * Handles the hash fragment response from public client code request. Returns a code response used by
-     * the client to exchange for a token in acquireToken.
-     * @param hashFragment
-     */
-    abstract handleFragmentResponse(hashFragment: string): CodeResponse;
+        // TODO expensive to resolve authority endpoints every time.
+        const authority: Authority = authorityString
+            ? AuthorityFactory.createInstance(authorityString, this.networkClient)
+            : this.defaultAuthorityInstance;
```

Comment:
Isn't Authority an abstract class? I don't think you can just instantiate Authority like this. Maybe throw an error instead?